tests/python_tests/test_generate_api.py

```diff
@@ -795,7 +794,144 @@ def test_chat_1(model_descr, generation_config):
+    assert chat_history_ov == chat_history_with_kv_cache
+
+from tokenizer_configs import get_tokenizer_configs
+chat_configs = [(k, v) for k, v in get_tokenizer_configs().items()]
+
+
+skipped_models = [
+    # These models fail even on HF so no need to check if applying chat matches.
+    "vibhorag101/llama-2-13b-chat-hf-phr_mental_therapy",
+    "codellama/CodeLlama-34b-Instruct-hf",
+    "deepseek-ai/deepseek-math-7b-rl",
+    "allenai/tulu-2-7b",
+    "alexsobolev/IcaroLM",
+    "tokyotech-llm/Swallow-7b-instruct-v0.1",
+    "bofenghuang/vigogne-2-7b-chat",
+    "OpenBuddy/openbuddy-mistral2-7b-v20.3-32k",
+    "AliAbdelrasheed/maqa_llama_4bit",
+    "stephenlzc/Mistral-7B-v0.3-Chinese-Chat-uncensored",
+
+    # Either ov_genai is unable to parse chat_template or results do not match with HF.
+    "meta-llama/Meta-Llama-3-8B-Instruct",
+    "databricks/dbrx-instruct",
+    "mosaicml/mpt-30b-chat",
+    "deepseek-ai/deepseek-coder-6.7b-instruct",
+    "maldv/winter-garden-7b-alpha",
+    "ishorn5/RTLCoder-Deepseek-v1.1",
+    "openchat/openchat-3.5-0106",
+    "casperhansen/llama-3-70b-instruct-awq",
+    "TheBloke/deepseek-coder-33B-instruct-GPTQ",
+    "AI-Sweden-Models/gpt-sw3-356m-instruct",
+    "google/gemma-7b-it",
+    "THUDM/cogvlm2-llama3-chat-19B",
+    "KnutJaegersberg/internlm-20b-llama",
+    "alpindale/WizardLM-2-8x22B",
+    "maywell/Synatra-Mixtral-8x7B",
+    "MediaTek-Research/Breeze-7B-Instruct-v1_0",
+    "bofenghuang/vigostral-7b-chat",
+    "meetkai/functionary-small-v2.5",
+    "nvidia/Llama3-ChatQA-1.5-8B",
+    "openchat/openchat-3.6-8b-20240522",
+    "tenyx/TenyxChat-7B-v1",
+    "LoneStriker/TinyLlama-1.1B-32k-Instruct-3.0bpw-h6-exl2",
+    "yam-peleg/Hebrew-Gemma-11B-V2",
+    "shenzhi-wang/Llama3-8B-Chinese-Chat",
+    "nlpai-lab/KULLM3",
+    "HuggingFaceH4/zephyr-7b-gemma-sft-v0.1",
+    "MediaTek-Research/Breeze-7B-Instruct-v0_1",
+    "shanchen/llama3-8B-slerp-biomed-chat-chinese",
+    "MLP-KTLim/llama-3-Korean-Bllossom-8B",
+    "lucyknada/microsoft_WizardLM-2-7B",
+    "aloobun/CosmicBun-8B",
+    "codellama/CodeLlama-70b-Instruct-hf",
+    "gorilla-llm/gorilla-openfunctions-v2",
+    "BramVanroy/Llama-2-13b-chat-dutch"
+]
+
+conversation = [
+    # {'role': 'system', 'content': 'You are a friendly bot.'},
+    {'role': 'user', 'content': '1+1='},
+    {'role': 'assistant', 'content': '1 + 1 = 2'},
+    {'role': 'user', 'content': 'What is the previous answer?'},
+    {'role': 'assistant', 'content': 'The previous answer was: 1 + 1 = 2'},
+    {'role': 'user', 'content': 'Why is the sun yellow?'},
+    {'role': 'assistant', 'content': 'Because it emits yeloow light.'},
+    {'role': 'user', 'content': 'What was my first question?'},
+    {'role': 'assistant', 'content': "I don't remember such difficult questions.\n Ask something different"},
+]
+@pytest.mark.precommit
+@pytest.mark.parametrize('chat_config', chat_configs)
+def test_apply_chat_template(model_tmp_path, chat_config):
```

Comment:
can we split big `tests/python_tests/test_generate_api.py` into files?
E.g. chat template is worth to have its dedicated file.

Reply:
Done. Split to 2 files. Common utilities are moved to `ov_genai_test_utils`